### ISSUE TITLE
Apache Webserver < 2.4.55 - CVE-2022-36760 & CVE-2022-37436

### ISSUE THREAD

**Issue by christiangierschner** [highmed/highmed-dsf#396]
Hi,

our scan revealed, that the dsf fhir-proxy uses a vulnerable apache version...
```
!---> Version: Apache/2.4.54 found in Container fhir-proxy_1 running OS_PRETTY_NAME="Alpine Linux v3.16" 
```

According to the changelog of the new 2.4.55, this affects the modules mod_proxy and mod_proxy_ajp...is this used in the fhir-proxy?

https://httpd.apache.org/security/vulnerabilities_24.html

**Comment by hhund** [comment]
Hi @christiangierschner,

thank you for the report. Here is our assessment on the [issues closed with Apache 2.4.55](https://httpd.apache.org/security/vulnerabilities_24.html):

> **moderate: mod_dav out of bounds read, or write of zero byte ([CVE-2006-20001](https://www.cve.org/CVERecord?id=CVE-2006-20001))**
> A carefully crafted If: request header can cause a memory read, or write of a single zero byte, in a pool (heap) memory location beyond the header value sent. This could cause the process to crash.

CVE-2006-20001 is not relevant for [DSF FHIR Proxy Version 0.9.0](https://github.com/highmed/highmed-dsf/pkgs/container/fhir_proxy/45206690?tag=0.9.0) since `mod_dav` is not being used.

<br>

> **moderate: Apache HTTP Server: mod_proxy_ajp Possible request smuggling ([CVE-2022-36760](https://www.cve.org/CVERecord?id=CVE-2022-36760))**
> Inconsistent Interpretation of HTTP Requests ('HTTP Request Smuggling') vulnerability in mod_proxy_ajp of Apache HTTP Server allows an attacker to smuggle requests to the AJP server it forwards requests to. This issue affects Apache HTTP Server Apache HTTP Server 2.4 version 2.4.54 and prior versions.

CVE-2022-36760 is not relevant for [DSF FHIR Proxy Version 0.9.0](https://github.com/highmed/highmed-dsf/pkgs/container/fhir_proxy/45206690?tag=0.9.0) since `mod_proxy_ajp` is not being used.

<br>

> **moderate: Apache HTTP Server: mod_proxy prior to 2.4.55 allows a backend to trigger HTTP response splitting ([CVE-2022-37436](https://www.cve.org/CVERecord?id=CVE-2022-37436))**
> Prior to Apache HTTP Server 2.4.55, a malicious backend can cause the response headers to be truncated early, resulting in some headers being incorporated into the response body. If the later headers have any security purpose, they will not be interpreted by the client.

CVE-2022-37436 is not relevant for [DSF FHIR Proxy Version 0.9.0](https://github.com/highmed/highmed-dsf/pkgs/container/fhir_proxy/45206690?tag=0.9.0). The module `mod_proxy` is being used, but the [DSF FHIR Server Version 0.9.0](https://github.com/highmed/highmed-dsf/pkgs/container/fhir/46110791?tag=0.9.0) is not a malicious backend and we are not sending any response headers with security purpose.

<br>

Based on this assessment there is no need to publish a new version before the next scheduled release.

**Comment by schwzr** [comment]
fixed in the new 0.9.1 release

https://github.com/highmed/highmed-dsf/releases/tag/v0.9.1
